src/core/Akka/Event/LoggingBus.cs

```diff
@@ -257,9 +256,12 @@ protected override bool Receive(object message)
 
             private static Debug ToDebug(UnhandledMessage message)
             {
+                // avoid NREs when we have ActorRefs.NoSender
+                var sender = Equals(message.Sender, ActorRefs.NoSender) ? "NoSender" : message.Sender.Path.ToString();
```

Comment:
Good alternative.

Reply:
We did something similar for `DeadLetter` logging - figured repurposing that was the way to go